source/server/overload_manager_impl.cc

```diff
@@ -92,29 +135,36 @@ OverloadAction::OverloadAction(const envoy::config::overload::v3::OverloadAction
   }
 
   active_gauge_.set(0);
+  scale_percent_gauge_.set(0);
 }
 
 bool OverloadAction::updateResourcePressure(const std::string& name, double pressure) {
-  const bool active = isActive();
+  const OverloadActionState old_state = getState();
 
   auto it = triggers_.find(name);
   ASSERT(it != triggers_.end());
-  if (it->second->updateValue(pressure)) {
-    if (it->second->isFired()) {
-      active_gauge_.set(1);
-      const auto result = fired_triggers_.insert(name);
-      ASSERT(result.second);
-    } else {
-      active_gauge_.set(0);
-      const auto result = fired_triggers_.erase(name);
-      ASSERT(result == 1);
+  if (!it->second->updateValue(pressure)) {
+    return false;
+  }
+  const auto trigger_new_state = it->second->actionState();
+  active_gauge_.set(trigger_new_state.isSaturated() ? 1 : 0);
+  scale_percent_gauge_.set(trigger_new_state.value() * 100);
+
+  {
```

Comment:
I think a comment somewhere would be useful here to help the reader understand why we need to iterate over all the triggers, it took me a minute to understand why this was necessary

Reply:
Done, thanks.